server/src/test/java/io/crate/metadata/ReferenceTest.java

```diff
@@ -125,36 +126,53 @@ public void test_streaming_of_reference_position_before_4_6_0() throws Exception
     @Test
     public void test_mapping_generation_for_varchar_with_length() throws Exception {
         SQLExecutor e = SQLExecutor.builder(clusterService)
-            .addTable("create table tbl (xs varchar(40))")
+            .addTable("create table tbl (xs varchar(40))", Settings.EMPTY, true)
             .build();
         DocTableInfo table = e.resolveTableInfo("tbl");
         Reference reference = table.getReference(new ColumnIdent("xs"));
         Map<String, Object> mapping = reference.toMapping();
         assertThat(mapping)
             .containsEntry("length_limit", 40)
             .containsEntry("position", 1)
+            .containsEntry("oid", 1L)
             .containsEntry("type", "keyword")
-            .hasSize(3);
+            .hasSize(4);
         IndexMetadata indexMetadata = clusterService.state().metadata().getIndices().valuesIt().next();
-        Map<String, Object> sourceAsMap = indexMetadata.mapping().sourceAsMap();
-        assertThat(Maps.getByPath(sourceAsMap, "properties.xs")).isEqualTo(mapping);
+        compareMappings(indexMetadata.mapping().sourceAsMap(), "xs", mapping);
     }
 
     @Test
     public void test_mapping_generation_for_string_without_doc_values() throws Exception {
         SQLExecutor e = SQLExecutor.builder(clusterService)
-            .addTable("create table tbl (xs string storage with (columnstore = false))")
+            .addTable("create table tbl (xs string storage with (columnstore = false))", Settings.EMPTY, true)
```

Comment:
Can actually revert changes of the whole file - I added OID generation for `CREATE TABLE` in AddColumnTask (as it was really important to pass validation) and this test started to fail. 

Then I made OID assignment optional, so this test will pass if I use .addTable without a flag. 
Should we keep it to highlight the OIDS presence? I would say no as we show OIDS in integration tests.